cmd/buildah/images.go

```diff
@@ -300,34 +306,63 @@ func outputImages(ctx context.Context, systemContext *types.SystemContext, store
 		}
 
 	outer:
-		for name, tags := range imageReposToMap(image.Names) {
-			for _, tag := range tags {
-				if !matchesReference(name+":"+tag, argName) {
+		for repo, tagsAndDigests := range imageReposToMap(image.Names) {
+			for _, tagOrDigest := range tagsAndDigests {
+				imageName := ""
+				imageDigest := ""
+				imageTag := ""
+				if repo != "" && tagOrDigest != "" {
+					if digest.Digest(tagOrDigest).Validate() == nil {
+						imageDigest = tagOrDigest
+						imageName = repo + "@" + imageDigest
+					} else {
+						imageTag = tagOrDigest
+						if len(image.Digest) > 0 {
+							imageDigest = image.Digest.String()
+						}
+						imageName = repo + ":" + imageTag
+					}
+				}
+
+				if !matchesReference(imageName, argName) {
 					continue
 				}
 				found = true
 
-				if !matchesFilter(ctx, store, image, name+":"+tag, filters) {
+				if !matchesFilter(ctx, store, image, imageName, filters) {
 					continue
 				}
+				imageDigests := make([]string, 0, len(image.Digests))
+				for _, imageDigest := range image.Digests {
+					imageDigests = append(imageDigests, imageDigest.String())
+				}
 				if opts.json {
-					jsonImages = append(jsonImages,
-						jsonImage{ID: image.ID,
-							Names:        image.Names,
-							Digest:       digest,
-							CreatedAtRaw: createdTime,
-							CreatedAt:    units.HumanDuration(time.Since((createdTime))) + " ago",
-							Size:         formattedSize(size),
-							ReadOnly:     image.ReadOnly,
-						})
+					var imageTags []string
+					for _, tagOrDigest := range tagsAndDigests {
```

Comment:
Should this really only list tags _from this single `repo` value_? The choice which particular repo will be used in the output is random, depending on how Go decides to order the map keys this time.

Reply:
That's a good reason to not add the `tags` field in this PR.  Dropping it.